Canyoneer/Canyoneer/AppDataKit/Persistence/UserPreferencesStorage+favorite.swift

```diff
@@ -6,40 +6,56 @@
 import Foundation
 
 extension UserPreferencesStorage {
-    
     fileprivate static let favoriteListKey = "canyon_favorite_list"
     public static let favorites = UserPreferencesStorage()
     
     static var allFavorites: [Canyon] {
-        let list = Self.favorites.get(key: Self.favoriteListKey) as [Canyon]?
-        return list ?? []
+        // Get legacy if exists
+        let pastList = ((favorites.get(key: favoriteListKey) as [LegacyCanyon]?) ?? []).map { Canyon(legacy: $0) }
+        
+        // Get current
+        let currentList = ((favorites.get(key: favoriteListKey) as [RopeWikiCanyon]?) ?? []).map { Canyon(data: $0) }
```

Comment:
This is doing the migration at line 24th but handling input from either source since we don't know if it has been migrated or not. In reality, I don't think merge actually does anything. We will get one or the other.